File: boot-data-couchbase/src/main/java/com/example/demo/WebConfig.java
package com.example.demo;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.server.RouterFunction;
import org.springframework.web.reactive.function.server.ServerResponse;

import static org.springframework.web.reactive.function.server.RequestPredicates.*;
import static org.springframework.web.reactive.function.server.RouterFunctions.route;

@Configuration
public class WebConfig {
    @Bean
    public RouterFunction<ServerResponse> routes(PostHandler handler) {
        return route(GET("/posts"), handler::all)
                .andRoute(POST("/posts"), handler::create)
                .andRoute(GET("/posts/{id}"), handler::get)
                .andRoute(PUT("/posts/{id}"), handler::update)
                .andRoute(DELETE("/posts/{id}"), handler::delete);
    }
}

File: kotlin-routes/src/main/kotlin/com/example/demo/WebConfig.kt
package com.example.demo

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.config.EnableWebFlux
import org.springframework.web.reactive.function.server.HandlerFunction
import org.springframework.web.reactive.function.server.RequestPredicates.*
import org.springframework.web.reactive.function.server.RouterFunctions.route
import org.springframework.web.reactive.function.server.ServerResponse
import org.springframework.web.reactive.function.server.RouterFunction


@Configuration
@EnableWebFlux
class WebConfig {

    @Bean
    fun routes(handler: PostHandler): RouterFunction<ServerResponse> {
        return route(GET("/posts"), HandlerFunction<ServerResponse>(handler::all))
                .andRoute(POST("/posts"), HandlerFunction<ServerResponse>(handler::create))
                .andRoute(GET("/posts/{id}"), HandlerFunction<ServerResponse>(handler::get))
                .andRoute(PUT("/posts/{id}"), HandlerFunction<ServerResponse>(handler::update))
                .andRoute(DELETE("/posts/{id}"), HandlerFunction<ServerResponse>(handler::delete))
    }
}